File: DevLogicCourse/src/Q03.java
import javax.swing.*;
import java.util.Scanner;
import java.util.HashMap;

public class Q03 {

    HashMap<String, String> agenda = new HashMap<>();

    public static void main(String[] args) {

        Q03 objAgenda = new Q03();

        Generar(objAgenda);
        boolean flag = true;
        while (flag) {

            Menu();
            Scanner capOption = new Scanner(System.in);
            System.out.print("Escoja la opción: ");
            int opc = capOption.nextInt();

            switch (opc) {
                case 1 -> Nuevo(objAgenda);
                case 2 -> Buscar(objAgenda);
                case 3 -> Editar(objAgenda);
                case 4 -> Eliminar(objAgenda);
                case 5 -> Listar(objAgenda);
                case 6 -> flag = false;
                default -> System.out.println("Opción no válida");
            }
        }
    }


    static void Generar(Q03 objAgenda) {
        objAgenda.agenda.put("Jonatan", "3005981009");
        objAgenda.agenda.put("Miguel", "3004451231");
        objAgenda.agenda.put("Amada", "3014454562");
        objAgenda.agenda.put("Elena", "3024451203");
        objAgenda.agenda.put("Ainara", "3114455694");
        objAgenda.agenda.put("Alanna", "3124458795");
        objAgenda.agenda.put("Cris", "3134459578");
        objAgenda.agenda.put("Yaireth", "3204547895");
        objAgenda.agenda.put("Poncho", "3214578495");
        objAgenda.agenda.put("Hilda", "3172237548");
    }


    static void Listar(Q03 objAgenda) {

        System.out.println("\nContactos\n");
        objAgenda.agenda.forEach((name, phone) -> System.out.printf("%s : %s\n", name, phone));

    }


    static void Menu() {
        System.out.println("\nContactos\n");
        System.out.println("1. Nuevo");
        System.out.println("2. Buscar");
        System.out.println("3. Editar");
        System.out.println("4. Eliminar");
        System.out.println("5. Listar");
        System.out.println("6. salir");
    }


    static void Nuevo(Q03 objAgenda) {
        Scanner capData = new Scanner(System.in);

        System.out.print("Nombre: ");
        String name = capData.nextLine();

        System.out.print("Teléfono: ");
        String phone = capData.nextLine();

        if (phone.matches("\\d+") && (phone.length() > 0 && phone.length() <= 11)) {
            objAgenda.agenda.put(name, phone);
        }
    }


    static void Buscar(Q03 objAgenda){
        Scanner capData = new Scanner(System.in);

        System.out.print("Nombre: ");
        String name = capData.nextLine();

        if (objAgenda.agenda.containsKey(name)){
            System.out.printf("%s : %s\n", name, objAgenda.agenda.get(name));
        }
        else {
            System.out.println("\nEl contacto no existe");
        }
    }


    static void Editar(Q03 objAgenda){

        Scanner capData = new Scanner(System.in);
        System.out.print("Nombre: ");
        String name = capData.nextLine();

        if (objAgenda.agenda.containsKey(name)){

            //Scanner capPhone = new Scanner(System.in);
            System.out.print("Nuevo número: ");
            String phone = capData.nextLine();

            if (phone.matches("\\d+") && (phone.length() > 0 && phone.length() <= 11)) {
                objAgenda.agenda.put(name, phone);
                System.out.println("\n¡Contacto actualizado!\n");
            }
            else {
                System.out.println("\nDigita sólo números\n");
            }
        }
        else {
            System.out.println("\nEl contacto no existe\n");
        }
    }


    static void Eliminar(Q03 objAgenda) {

        Scanner capName = new Scanner(System.in);
        System.out.print("Nombre: ");
        String name = capName.nextLine();

        if (objAgenda.agenda.containsKey(name)){
            objAgenda.agenda.remove(name);
            System.out.println("\nContacto eliminado\n");
        }
        else {
            System.out.println("\nEl contacto no existe\n");
        }
    }
}

File: Python/03 - Estructuras de datos/Q03.py

contact: dict = {
    "Miguel" : "3004451231",
    "Amada" : "3014454562",
    "Elena" : "3024451203",
    "Ainara" : "3114455694",
    "Alanna" : "3124458795",
    "Cris" : "3134459578",
    "Yaireth" : "3204547895",
    "Poncho" : "3214578495",
    "Hilda" : "3172237548"
}

def insert_phone(name):
        phone = input("Teléfono: ")
        if phone.isdigit() and (len(phone) > 0 and len(phone) <= 11):
            contact[name] = phone
        else:
            print("El teléfono debe ser numérico entre uno y 11 dígitos.")


def new():
    name = input("Nombre: ")
    insert_phone(name)


def find():
    name = input("Nombre del contacto: ")
    if name in contact: print(f"Nombre: {name} | Teléfono: {contact[name]}.")
    else: print(f"El contacto \"{name}\" no existe.")


def update():
    name = input("Introduce el nombre del contacto a actualizar: ")
    if name in contact: insert_phone(name)
    else: print(f"El contacto \"{name}\" no existe.")


def delete():
    name = input("Introduce el nombre del contacto a eliminar: ")
    if name in contact: del contact[name]
    else: print(f"El contacto {name} no existe.")


def listing():
    for name in contact:
        print(f'{name:<15} : {contact[name]:<12}')


def menu():
    print('\n***** MENU *****\n')
    print('1. Nuevo')
    print('2. Buscar')
    print('3. Editar')
    print('4. Eliminar')
    print('5. Listar')
    print('0. Salir\n')


while True:
    
    menu()
    opc = input('Digita la opción: ')
    print('')
    
    match opc:
        case '1': new()
        case '2': find()
        case '3': update()
        case '4': delete()
        case '5': listing()
        case '0': break
        case _: print('¡¡¡ Opción errada !!!')
